osu.Framework/Audio/Track/TrackBass.cs

```diff
@@ -303,6 +303,33 @@ public override async Task<bool> SeekAsync(double seek)
             return conservativeClamped == seek;
         }
 
+        protected override Tags? GetTags()
+        {
+            var tags = TagReader.Read(activeStream);
+
+            if (tags == null)
+                return null;
+
+            var parsed = new Tags
+            {
+                Artist = tags.Artist?.Trim()!,
+                Title = tags.Title,
+                Album = tags.Album,
+                Genre = tags.Genre,
+            };
+
+            if (string.IsNullOrEmpty(parsed.Artist))
+                parsed.Artist = tags.AlbumArtist;
+
+            if (Int32.TryParse(tags.BPM, out int bpm))
```

Comment:
```suggestion
            if (int.TryParse(tags.BPM, out int bpm))
```

Also what's the guarantee that this is gonna parse as `int`? What about fractional BPM?

Reply:
Hadn't considered that, on second thought not even sure this is worth having because it doesn't seem to be populated very commonly.